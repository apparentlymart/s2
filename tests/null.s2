
layerinfo "type" = "core";
layerinfo "majorversion" = "1";

function builtin string(int n) : string;

class TestClass {
    var string member;
}

function returnThingy() : TestClass {
    return null TestClass;
}

function main() {
    var TestClass test;
    var bool nullness = isnull $test;
    println ((isnull $test) ? "It's null" : "NOT A NULL!");
    if (isnull $test) {
        println "It's still null.";
    }
    if (isnull returnThingy()) {
        println "The function returned a null thingy";
    }
    $test = new TestClass;
    println "Member = \"$test.member\"";
}
