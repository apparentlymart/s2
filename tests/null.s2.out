It's null
It's still null.
The function returned a null thingy
Member = ""
